### ISSUE TITLE
AttributeError: 'my_spawn' object has no attribute 'ptyproc'

### ISSUE THREAD

**Issue by hundeboll** [pexpect/pexpect#298]
We subclass `spawn` from `pty_spawn.py` to defer expecting on commands through a telnet connection. In this case, the command is not know at init-time, and thus `command` is `None`, which in turn avoids running `_spawn()` in the super class.

In this setup, a timeout exception tries to stringify the `spawn` class, which reads the `flag_eof` property, where `self.ptyproc` is accessed, but doesn't exist (due to not calling `_spawn()`, leading to an uncaught exception:
```
Traceback (most recent call last):
  File "./except.py", line 19, in <module>
    child.expect('foo', 1)
  File "/usr/lib/python2.7/site-packages/pexpect/spawnbase.py", line 315, in expect
    timeout, searchwindowsize, async)
  File "/usr/lib/python2.7/site-packages/pexpect/spawnbase.py", line 339, in expect_list
    return exp.expect_loop(timeout)
  File "/usr/lib/python2.7/site-packages/pexpect/expect.py", line 95, in expect_loop
    return self.timeout()
  File "/usr/lib/python2.7/site-packages/pexpect/expect.py", line 65, in timeout
    msg = str(spawn)
  File "/usr/lib/python2.7/site-packages/pexpect/pty_spawn.py", line 213, in __str__
    s.append('flag_eof: ' + str(self.flag_eof))
  File "/usr/lib/python2.7/site-packages/pexpect/pty_spawn.py", line 569, in flag_eof
    return self.ptyproc.flag_eof
AttributeError: 'my_spawn' object has no attribute 'ptyproc'
```

The exception is reproducible with the following code:
```
 #!/usr/bin/env python2

import pexpect

class my_spawn(pexpect.spawn):
    def __init__(self, command=None):
        self.args = None
        self.command = command
        pexpect.spawn.__init__(self, None, None, timeout=1)

        self.child_fd = -1
        self.own_fd = False
        self.closed = False

    def read_nonblocking (self, size=1, timeout=-1):
        return ""

child = my_spawn('sleep 10')
child.expect('foo', 1)
```

**Comment by jquast** [comment]
In the mean time, you can derive and modify ``__str__`` to exclude/fix such issue, but this is a relatively easy issue to resolve upstream, thank you very much for reporting!

**Comment by jquast** [comment]
fix proposed in #310, awaiting review
